libraries/AP_Scripting/applets/arming-checks.lua

```diff
@@ -0,0 +1,441 @@
+        if q_options ~= nil then
+            local q_land_option = ((q_options & (1 << 5)) > 0)
+            local q_RTL_option = ((q_options & (1 << 20)) > 0)
+
+            if (not q_land_option) and (not q_RTL_option) then
+                return false
+            end
+        end
+    end
+    return true -- always ok if not a QuadPlane
+end
+local function qrtl_warning()
+    if Q_ENABLE:get() > 0 then
+        if q_options ~= nil then
+            local q_land_option = ((q_options & (1 << 5)) > 0)
+            local q_RTL_option = ((q_options & (1 << 20)) > 0)
+            if q_land_option or q_RTL_option then
+                return false
+            end
+        end
+    end
+    return true -- always ok if not a QuadPlane
+end
+
+-- A single check to ensure AIRSPEED_STALL < AIRSPEED_MIN < AIRSPEED_CRUISE < AIRSPEED_MAX
+local function airspeed_check()
+    local airspeed_stall = AIRSPEED_STALL:get() or -1
+    local airspeed_min = AIRSPEED_MIN:get() or -1
+    local airspeed_cruise = AIRSPEED_CRUISE:get() or -1
+    local airspeed_max = AIRSPEED_MAX:get() or -1
+
+    if airspeed_stall > 0 then
+        if airspeed_min < airspeed_stall then
+            return false
+        end
+    end
+    if airspeed_cruise < airspeed_min then
+        return false
+    end
+    if airspeed_max < airspeed_cruise then
+        return false
+    end
+    return true
+end
+
+-- If AIRSPEED_STALL is set AIRSPEED_MIN should be at least 25% higher than AIRSPEED_STALL
+local function stallspeed_check()
+    local airspeed_stall = AIRSPEED_STALL:get() or -1
+    local airspeed_min = AIRSPEED_MIN:get() or -1
+
+    if airspeed_stall > 0 then
+        if airspeed_min > (airspeed_stall * 1.25) then
+            return false
+        end
+    end
+
+    return true
+end
+
+-- From the wiki: This is the center of the speed scaling range and should be set close to the normal cruising speed of the vehicle.
+local function scaling_speed_check()
+    local scaling_speed = SCALING_SPEED:get() or -1
+    local airspeed_cruise = AIRSPEED_CRUISE:get() or -1
+    if scaling_speed <= 0 or airspeed_cruise <= 0 then
+        return false
+    end
+    -- use a 20% range for "close"
+    if scaling_speed < (airspeed_cruise * 0.8) or scaling_speed > (airspeed_cruise * 1.2) then
+        return false
+    end
+    -- if either are not set then
+    return true
+end
+
+-- Arming checks can be deleted if not required, or set the parameter to MAV_SEVERITY.NONE to avoid changing the script
+-- or new arming checks added 
+local arming_checks = {
+    -- __call = function(cls, func, param_name, text, severity, passed, changed) -- constructor
+--[[
+    // @Param: ZAR_SYSID
+    // @DisplayName: MAV_SYSID must be set
+    // @Description: Check that MAV_SYSID (or SYDID_THISMAV) has been set. 3 or less to prevent arming. -1 to disable.
+    // @Values: 0:Emergency(PreArm),1:Alert(PreArm),2:Critical(PreArm),3:Error(PreArm),4:Warning,5:Notice,6:Info,7:Debug,-1:Disabled
```

Comment:
maybe we should put -1 at the front of the list so it's in numerical order?

Reply:
Well I am hoping to simply extend it to do other vehicles, rather than having different scripts. I'm expecting there will be a lot of common checks. That's my theory anyway.